docs/root/statistics.md

```diff
@@ -0,0 +1,121 @@
+[#391](https://github.com/envoyproxy/nighthawk/pull/391)).
+
+Name | Type | Description
+-----| ----- | ----------------
+upstream_rq_total | Counter | Total number of requests sent from Nighthawk
+http_1xx | Counter | Total number of response with code 1xx
+http_2xx | Counter | Total number of response with code 2xx	
+http_3xx | Counter | Total number of response with code 3xx	
+http_4xx | Counter | Total number of response with code 4xx	
+http_5xx | Counter | Total number of response with code 5xx	
+http_xxx | Counter | Total number of response with code <100 or >=600
+stream_resets | Counter | Total number of stream reset	
+pool_overflow | Counter | Total number of times connection pool overflowed	
+pool_connection_failure | Counter | Total number of times pool connection failed	
+benchmark_http_client.latency_1xx | HdrStatistic | Latency (in Nanosecond) histogram of request with code 1xx	
+benchmark_http_client.latency_2xx | HdrStatistic | Latency (in Nanosecond) histogram of request with code 2xx
+benchmark_http_client.latency_3xx | HdrStatistic | Latency (in Nanosecond) histogram of request with code 3xx	
+benchmark_http_client.latency_4xx | HdrStatistic | Latency (in Nanosecond) histogram of request with code 4xx
+benchmark_http_client.latency_5xx | HdrStatistic | Latency (in Nanosecond) histogram of request with code 5xx	
+benchmark_http_client.latency_xxx | HdrStatistic | Latency (in Nanosecond) histogram of request with code <100 or >=600
+benchmark_http_client.queue_to_connect | HdrStatistic | Histogram of request connection time	(in Nanosecond)
+benchmark_http_client.request_to_response | HdrStatistic | Latency (in Nanosecond) histogram of all requests (include requests with stream reset or pool failure)
+benchmark_http_client.response_header_size | StreamingStatistic | Statistic of response header size (min, max, mean, pstdev values in bytes)
+benchmark_http_client.response_body_size | StreamingStatistic | Statistic of response body size (min, max, mean, pstdev values in bytes)
+sequencer.callback | HdrStatistic | Latency (in Nanosecond) histogram of unblocked requests
+sequencer.blocking | HdrStatistic | Latency (in Nanosecond) histogram of blocked requests
+
+
+## Envoy Metrics Model
+
+[Envoy](https://github.com/envoyproxy/envoy) has 3 types of metrics	
+- Counters: Unsigned integers (can only increase) represent how many times an
+  event happens, e.g. total number of requests.	
+- Gauges: Unsigned integers (can increase or decrease), e.g. number of active
+  connections.	
+- Histograms: Unsigned integers that will yield summarized percentile values.
+  E.g. latency distributions.	
+
+In Envoy, the stat
+[Store](https://github.com/envoyproxy/envoy/blob/74530c92cfa3682b49b540fddf2aba40ac10c68e/include/envoy/stats/store.h#L29)
+is a singleton and provides a simple interface by which the rest of the code can
+obtain handles to scopes, counters, gauges, and histograms. Envoy counters and
```

Comment:
`Store` is not a type of metrics so I decide not to list its definition above with other metrics. Since it is a (common) Envoy concept and its definition is not closely relevant in the NH Statistics doc, so only a link to the code definition is provided here.

Reply:
Sorry I was unclear here.  I was actually referring to scopes.   I realize that scopes are probably not metrics themselves, but I'm not sure what they are in this context, and unlike store, we don't have a link to envoy documentation.

Reply:
Added a code link to scopes too.